Sources/FlickTypeKit/FlickTypeKit-SwiftUI.swift

```diff
@@ -38,6 +40,7 @@ public struct FlickTypeTextEditor: View {
         startingText: self.text) { items in
           if let newText = items?.first as? String {
             self.text = newText
+            onCommit()
```

Comment:
I'm presuming it makes most sense for this to only run the handler if there is valid text, but not 100% sure on that.

Reply:
This is a great addition, thanks!

FlickType actually has a somewhat obscure API that would be a perfect fit for this: you can distinguish the `FlickType.CompletionType` between:
- `.dismiss`: The user swiped down from the top of the screen to dismiss the keyboard, and
- `.action`: The user tapped the “Done” button

So I think it’s best to only fire `onCommit()` when the user has pressed the “Done” button but not when the keyboard has been dismissed, like so:

```
if items?.completionType == .action {
  onCommit()
}
```

So in a messaging app, for example, you might only send the actual message inside your `onCommit()` handler so that if the user wants to preview what they’ve typed so far, they can just swipe down to see their message and even tap to edit it further until they’re satisfied and either hit “Done” from the keyboard or use some action button from your app.
 
(Note: This API is currently implemented as an awkward `.completionType` extension on `Array` in order to preserve the signature of the default completion handler, and I’ll probably change it in the future. But you don’t have to worry about this: if and when I do, I’ll make the change here as well.)

Reply:
Oh nice! Updated still inside of the original `if` block. Should I update the outer if to a `guard` statement, might be a bit cleaner to read that way?

Reply:
Awesome! I’m ready to merge this now, but if you’d like to make it even cleaner please go ahead, by all means! :)

Reply:
Done! 😊